chopper/lib/src/request.dart

```diff
@@ -45,23 +45,27 @@ class Request extends http.BaseRequest {
   /// the [parameters] are merged together.
   Request.uri(
     String method,
-    Uri url, {
+    Uri url,
+    String baseUrl, {
     this.body,
-    Map<String, dynamic>? parameters,
     Map<String, String> headers = const {},
     this.multipart = false,
     this.parts = const [],
     this.useBrackets = false,
     this.includeNullQueryVars = false,
-  })  : origin = url.origin,
+  })  : origin = url.isScheme('HTTP') || url.isScheme('HTTPS')
+            ? url.origin
+            : baseUrl,
         path = url.path,
-        parameters = {...url.queryParametersAll, ...?parameters},
```

Comment:
Hmm, I kept this here specifically for edge cases when one wants to add additional parameters to the Uri without messing with the Uri itself.

Reply:
I think it could be confusing having multiple ways of supplying parameters. Of course it wouldn't hurt to keep both. But even if someone doesn't want to mess with the orginal Uri,  one could use `Uri.replace` which creates a new instance and leaving the old intact.  An egde case like that probably isn't hard to work around.

But maybe I don't really see the edge case right.

Reply:
The [Uri.queryParameters](https://github.com/dart-lang/sdk/blob/stable/sdk/lib/core/uri.dart#L212) only accepts a `Map` containing `String` or `Iterable<String>`. 

```dart
// this compiles fine
final Uri url = Uri.https(
  'chopper.dev',
  '/test',
  {
    'foo': 'bar',
    'foo_list': [
      'one',
      'two',
      'three',
    ],
  },
);
```

If you try to use a `Map<String, String>` it will throw an exception

`type '_InternalLinkedHashMap<String, String>' is not a subtype of type 'Iterable<dynamic>'`

```dart
// this does not compile
final Uri url= Uri.https(
  'chopper.dev',
  '/test2',
  {
    'foo': 'bar',
    'foo_list': [
      'one',
      'two',
      'three',
    ],
    'user': {
      'name': 'john',
      'surname': 'doe',
    },
  },
);
```

Chopper's `Request.parameters`, however, will accept a `Map<String, dynamic>`.

```dart
test('works', () {
  expect(
    Request.uri(
      'GET',
      Uri.parse(
        'https://chopper.dev/test3',
      ),
      parameters: {
        'foo': 'bar',
        'foo_list': [
          'one',
          'two',
          'three',
        ],
        'user': {
          'name': 'john',
          'surname': 'doe',
        },
      },
    ).url.toString(),
    equals(
      'https://chopper.dev/test3?'
      'foo=bar'
      '&foo_list=one'
      '&foo_list=two'
      '&foo_list=three'
      '&user.name=john'
      '&user.surname=doe',
    ),
  );
});
```

Chopper has had this [feature](https://github.com/lejard-h/chopper/blob/develop/chopper/lib/src/utils.dart#L90) for a while and people might rely on it.

On top of that, the getters [Uri.queryParameters](https://github.com/dart-lang/sdk/blob/stable/sdk/lib/core/uri.dart#L4521) and [Uri.queryParametersAll](https://github.com/dart-lang/sdk/blob/stable/sdk/lib/core/uri.dart#L4526) both parse the already encoded query string to get the parameters back into a `Map<String, dynamic>` which means that the original `Map<String, dynamic>` that it passed to the `Uri` constructor is not preserved.

That's why [I merged both](https://github.com/lejard-h/chopper/blob/develop/chopper/lib/src/request.dart#L58) before sending the final `parameters` to [buildUri](https://github.com/lejard-h/chopper/blob/develop/chopper/lib/src/request.dart#L102).

Reply:
Aah makes more sense now. I will revert the removal of the parameters for now.

Reply:
Cool. 😎 While you're at it, please dump that test case above into [chopper/test/request_test.dart](https://github.com/lejard-h/chopper/blob/develop/chopper/test/request_test.dart). 😇